### ISSUE TITLE
Suggestion: could this plugin act like code_complete

### ISSUE THREAD

**Issue by snorez** [tenfyzhong/CompleteParameter.vim#9]
Awesome. It works well.

an example(C source):
`printf|` and trigger ycm, then it shows
`printf( const char *restrict __format, ... )`.
then we do parameter_complete,
`printf(constchar*__format, ...)` with SELECT-MODE on `constchar*__format`.
could this be something like
```c
printf(`<const char *restrict __format>`, `<...>`)
```
'\`<' and '>\`' are the keywords we could use to search and jump.

**Comment by tenfyzhong** [comment]
It works like what you said. But I have remove the type declaration from the parameters. When the parameter completed, the text `printf(char*, ...)` will be inserted, and then select parameter `char*`(select mode). You can type anything to instead the parameter. And then type `<m-n>`, it will jump to `...` and select it. Here's a screenshot.

[screenshot](https://ws4.sinaimg.cn/large/006tNc79ly1fh3isghf3qg30hs0dcqn2.gif)

The picture is to large, it can't put in the markdown.

**Comment by snorez** [comment]
@tenfyzhong  yeah, I think it would be wonderfull to show the type that the argument is.
`printf` shows
```
printf( *char, ... )
```
no type inclusive.
I found that when we edit an argument, it still could jump back to it.
```
printf("%s\n", "hello world");
```
then <m-p>, it back to `"hello world"`.
but it can only move back to `"hello world"`.

Also, if we don't modify the argment, could it move back to the first one when it hit the last one?

**Comment by tenfyzhong** [comment]
I don't think there's a good idea to include the type of the parameter. We can always know the type from the parameter name. 
It can not jump back to the first parameter, there's a bug. Thanks to report it. I will fix it later.

**Comment by tenfyzhong** [comment]
By the way, sometimes I have a variable has the same name of the parameter, I no need to do anything, just goto the next parameter.

**Comment by snorez** [comment]
> By the way, sometimes I have a variable has the same name of the parameter, I no need to do anything, just goto the next parameter.


I don't know the meaning. Can you take some examples?

**Comment by tenfyzhong** [comment]
for example:
![](https://ws2.sinaimg.cn/large/006tNc79ly1fh3n695b2qg30hs0dcgrx.gif)
You have two variable, num1 and num2, and use them to call the add function. 
After the completion, it'll select the `num1`, but you no need to modify it, you only need to do is jump to the next parameter.

**Comment by tenfyzhong** [comment]
>then , it back to "hello world".
>but it can only move back to "hello world".

>Also, if we don't modify the argment, could it move back to the first one when it hit the last one?

I have fix this bug by the commit `08cb954`.
